### PR TITLE
Implement buyFromGallery function with tests

### PR DESCRIPTION
**Pull request by katelynsills**
The gallery userFacet now has a function called `buyFromGallery` which allows the user to pass in a pixelAmount that specifies their desired pixels. If someone has previously sold those pixels to the gallery, the gallery prices the pixels, and instantiates an escrow smart contract where the user is able to pay the price to receive the pixels.

### CODE REVIEW

File: demo/gallery/vat-alice.js [hunk 1]
```diff
@@ -216,16 +216,34 @@ function makeAliceMaker(E, log) {
           // willing to offer for it
           // sellToGallery returns an invite to the smart contract
           const { inviteP, host } = await E(gallery).sellToGallery(amount);
-          const seatP = E(host).redeem(inviteP);
-          E(seatP).offer(exclusivePixelPaymentP);
-          const dustPurseP = E(dustIssuer).makeEmptyPurse();
-          const pixelPurseP = E(pixelIssuer).makeEmptyPurse();
-          return E(gallery).collectFromGallery(
+          const seatP = await E(host).redeem(inviteP);
+          await E(seatP).offer(exclusivePixelPaymentP);
```
Comment (Chris-Hibbert) [inline demo/gallery/vat-alice.js]: As I understand it, the await on 219 means that seatP should be 'seat', and you don't need E(seatP).

File: demo/gallery/vat-alice.js [hunk 2]
```diff
@@ -216,16 +216,34 @@ function makeAliceMaker(E, log) {
...
+          const dustPurseP = await E(dustIssuer).makeEmptyPurse();
+          const pixelPurseP = await E(pixelIssuer).makeEmptyPurse();
```
Comment (Chris-Hibbert) [inline demo/gallery/vat-alice.js]: These two awaits are unnecessary since they're always used as promises below.

File: demo/gallery/vat-alice.js [hunk 3]
```diff
@@ -216,16 +216,34 @@ function makeAliceMaker(E, log) {
...
+          // now buy it back
+          const {
+            inviteP: buyBackInviteP,
+            host: buyBackHost,
+            dustNeeded,
+          } = await E(gallery).buyFromGallery(amount);
+          const buyBackSeatP = await E(buyBackHost).redeem(buyBackInviteP);
+          const dustPaymentP = await E(dustPurseP).withdraw(dustNeeded);
+
+          await E(buyBackSeatP).offer(dustPaymentP);
```
Comment (Chris-Hibbert) [inline demo/gallery/vat-alice.js]: unnecessary await.

File: demo/gallery/vat-alice.js [hunk 4]
```diff
@@ -216,16 +216,34 @@ function makeAliceMaker(E, log) {
...
+          // alice is buying a pixel, so her win purse is a pixel
+          // purse and her refund purse is a dust purse
+          await E(gallery).collectFromGallery(
+            buyBackSeatP,
+            pixelPurseP,
+            dustPurseP,
+            'alice escrow 2',
+          );
```
Comment (Chris-Hibbert) [inline demo/gallery/vat-alice.js]: This would be a good place to log the values of the purses, wouldn't it?

File: more/pixels/gallery.js [hunk 5]
```diff
@@ -284,13 +284,17 @@ export function makeGallery(
...
   // only pixels can be sold to the gallery, not use or transfer rights
   function sellToGallery(pixelAmountP) {
     return Promise.resolve(pixelAmountP).then(async pixelAmount => {
       pixelAmount = pixelAssay.coerce(pixelAmount);
       const dustAmount = pricePixelAmount(pixelAmount);
-      const dustPurse = await dustMint.mint(dustAmount);
-      const dustPaymentP = await dustPurse.withdraw(
+      // just mint the dust that we need
+      const tempDustPurseP = await dustMint.mint(dustAmount);
+      const dustPaymentP = await tempDustPurseP.withdraw(
```
Comment (Chris-Hibbert) [inline more/pixels/gallery.js]: These awaits aren't useful, either.

File: more/pixels/gallery.js [hunk 6]
```diff
@@ -305,18 +309,60 @@ export function makeGallery(
...
+      }
+      const pixelPaymentP = await E(sellBuyPixelPurseP).withdraw(pixelAmount);
+      const dustAmount = pricePixelAmount(pixelAmount);
+
+      // same order as in sellToGallery
+      // the left will have to provide dust, right will have to
+      // provide pixels. Left is the user, right is the gallery
+      const terms = harden({ left: dustAmount, right: pixelAmount });
+      const contractHost = makeContractHost(E, evaluate);
+      const escrowExchangeInstallationP = await E(contractHost).install(
```
Comment (Chris-Hibbert) [inline more/pixels/gallery.js]: await not needed and on 347

File: demo/gallery/vat-alice.js (hunk 4 above)
Comment (katelynsills) [reply demo/gallery/vat-alice.js]: Good idea

**Pull request merged**